app/queries/servers/post.ts

```diff
@@ -279,3 +279,42 @@ export const countUsersFromMentions = async (database: Database, mentions: strin
     const [groups, usersCount] = await Promise.all([groupsQuery, usersQuery]);
     return groups.reduce((acc, v) => acc + v.memberCount, usersCount);
 };
+
+export const queryPostsWithPermalinkReferences = async (
+    database: Database,
+    referencedPostId: string,
+    channelId?: string,
+): Promise<PostModel[]> => {
+    try {
+        const clauses: Q.Clause[] = [
+            Q.where('metadata', Q.notEq(null)),
+            Q.where('delete_at', Q.eq(0)),
```

Comment:
I think elsewhere you are checking that delete_at > 0 to determine if the post was deleted, but that will never be true as this is excluding deleted posts.